azure/scope/identity.go

```diff
@@ -159,6 +161,13 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = p.cache.GetOrStoreManagedIdentity(&options)
 
+	case infrav1.UserAssignedIdentityCredential:
+		cloudType := parseCloudType(p.Identity.Spec.UserAssignedIdentityCredentialsCloudType)
+		cred, authErr = dataplane.NewUserAssignedIdentityCredential(ctx, p.Identity.Spec.UserAssignedIdentityCredentialsPath, dataplane.WithClientOpts(azcore.ClientOptions{Cloud: cloudType}))
+		if authErr != nil {
+			return nil, authErr
+		}
+
```

Comment:
Are you able to add some test cases to `TestGetTokenCredential` to cover this new flow?